packages/intersection-observer/src/index.ts

```diff
@@ -57,31 +57,97 @@ export type E = JSX.Element;
  * <div use:observer></div>
  * ```
  */
-export const createIntersectionObserver = (
-  elements: MaybeAccessor<Element[]>,
+export const makeIntersectionObserver = (
+  elements: Element[],
   onChange: IntersectionObserverCallback,
   options?: IntersectionObserverInit
-): [
-  AddIntersectionObserverEntry,
-  {
-    remove: RemoveIntersectionObserverEntry;
-    start: () => void;
-    stop: () => void;
-    instance: IntersectionObserver;
-  }
-] => {
+): {
+  add: AddIntersectionObserverEntry,
+  remove: RemoveIntersectionObserverEntry;
+  start: () => void;
+  reset: () => void;
+  stop: () => void;
+  instance: IntersectionObserver;
+} => {
   const instance = new IntersectionObserver(onChange, options);
   const add: AddIntersectionObserverEntry = el => instance.observe(access(el));
   const remove: RemoveIntersectionObserverEntry = el => instance.unobserve(access(el));
-  const start = () => access(elements).forEach(el => add(el));
+  const start = () => elements.forEach(el => add(el));
   const stop = () => instance.disconnect();
+  const reset = () => instance.takeRecords().forEach((el) => remove(el.target));
   onMount(start);
   onCleanup(stop);
-  return [add, { remove, start, stop, instance }];
+  return { add, remove, start, stop, reset, instance };
+};
+
+/**
+ * Creates a reactive Intersection Observer primitive.
+ *
+ * @param elements - A list of elements to watch
+ * @param onChange - An event handler that returns an array of observer entires
+ * @param options - IntersectionObserver constructor options:
+ * - `root` — The Element or Document whose bounds are used as the bounding box when testing for intersection.
+ * - `rootMargin` — A string which specifies a set of offsets to add to the root's bounding_box when calculating intersections, effectively shrinking or growing the root for calculation purposes.
+ * - `threshold` — Either a single number or an array of numbers between 0.0 and 1.0, specifying a ratio of intersection area to total bounding box area for the observed target.
+ *
+ * @example
+ * ```tsx
+ * const createIntersectionObserver(els, entries =>
+ *   console.log(entries)
+ * );
+ * ```
+ */
+export const createIntersectionObserver = (
+  elements: MaybeAccessor<Element[]>,
+  onChange: IntersectionObserverCallback,
+  options?: IntersectionObserverInit
+) => {
+  const { add, reset } = makeIntersectionObserver([], onChange, options);
+  createEffect(() => {
```

Comment:
Effect could be created only if the `elements` arg in an accessor.

Reply:
Well if we do that it's just basically a `makeIntersectionObserver` so I've changed it to force an Accessor since that's the only thing that makes sense here. I feel like it's so stripped down now but perhaps this is indeed the basic reactive use?

Reply:
That would be the point in using it I think.

Reply:
Yes yes, lol. I agree.